scipy/sparse/linalg/_dsolve/linsolve.py

```diff
@@ -331,7 +333,7 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
 
     if not isspmatrix_csc(A):
         A = csc_matrix(A)
-        warn('splu requires CSC matrix format', SparseEfficiencyWarning)
+        warn('splu() converted its input to CSC format', SparseEfficiencyWarning)
```

Comment:
It looks like we have a bunch of tests that were expecting the old warning message, that are now failing with the new message. Can you update these tests to the new messages please?